### ISSUE TITLE
There's an issue with my head.. it hurts..

### ISSUE THREAD

**Issue by Malreckus** [DarthAffe/RGBSyncPlus#18]
Hi!

I see that there are several issues regarding "device" here and I think I have a possible solution to issues like these: We're noobs and we need help.

All kinds of people who want their RGB to be synced can find this project through google. That includes people like me who are spoiled and expects an executable file to work exactly how they imagine it will work.

I've been sitting here for over an hour trying to understand how I can get your program to find my devices. I've looked through your replies in the different issues, and I think the answer to my question is that I need to learn how to "build"..? 

I wasn't able to find a compatibility list of supported devices. The RGB devices I have are:
* Keyboard - Razer Ornata
* Mouse - Logitech G703
* Mousepad - Logitech G Powerplay
* CPU Cooler - Corsair H115i (No RGB on fans)
* RAM - G.Skill Trident Z RGB
* Motherboard - ASUS Prime Z370-A + LED strip
* Philips HUE lights (3 bulbs and one strip) (Not expecting this to be possible)
* Cheap LED strip from ebay, which is powered through USB (Absolutely not expecting this)

This is what I have done - after learning from some mistakes:
1. Downloaded latest release
2. Unblocked .zip file and extracted
3. Copied the ASUS, Logitech, Razer and Corsair .dll files from "DeviceProvider" and pasted in "x86"
- I'm guessing that copy/pasting like this wasn't what you meant in other replies..

I first got access to my non-RGB GPU (RTX 2080). After copy/pasting dll files, I got access to the Corsair cooler and a list of five components called motherboard. I tried syncing the components with the cooler, but that crashed the program and I've not been able to find the motherboard devices again.

Now.. what do I do, hehe. I think it's such a cool thing you're doing and I'm not expecting you to guide me through every step. But could you give me (and others like me) some pointers?
Maybe you can tell me what I have to learn in order to edit/adjust (or is this what you call "build"?) whatever's needed to fix my problems?

Or is it just I who have misunderstood everything?

I apologize for bothering you with this and I will await your reply patiently!

**Comment by DarthAffe** [comment]
Hey, first of all to clarify that point a bit: You don't need how to build, nor do you need to copy anything. This would only be needed if you want to change something on your own. The release-zip has this already done.

The main problem right now regarding the things you described is the fact, that for me (and quite some other people too) it just works. I double click the exe and everything is fine. But I'm well aware of the fact, that that's (sadly) not for everyone the case. Especially the aura-SDK has some nasty quirks/bugs which can cause various problems depending on the available devices, the actions taken and so on.

The general steps (that should work) are:
1. Download the zip from releases
2. (On Win8 and newer:) Right click, go to properties, unblock the zip
3. Extract the zip
4. Run the .exe (you might need to run it as administrator on Win8 and newer)

After that you should at least have it running in the tray.
Now for device detection: 
- **Corsair** should always work out of the box but since you don't have a RGB device it's obviously nothing shown.
- **Logitech** is a bit special in how the SDK handles devices. Since there's no built in way to detect devices it depends on the HIDs to be mapped in the library itself. The G703 and the mousepad aren't included right now - that's why they aren't detected. If you're able to look up the HIDs (in the device manager) of these devices I can add them.
- **Razer** is also a bit special but more in the way that I don't have access to any razer device myself and I don't know many people to work with on that who own one. Also the way the SDK handles different devices is quite different so it might be that it just doesn't work for some devices while the tested ones are working. My expectation is "Razer just works out of the box" but for the reasons above I can't promise that and I won't be too surprised if it isn't for some devices.
- **Asus** While being the most used/requested feature it's the messiest one too (but compared to other mainboard brands at least it works somehow). Like i said above aura is a bit ... special ... memory isn't supported even if it's implemented (it seems like they don't want to support non asus-devices in the SDK, at least that's what the forum post about it sounded like). That's why most people use aura as master to sync too. There're also quite some devices reported in a wrong way (which can cause the crash). I can't really help with that aside from the advice that if it crashes add one device/led at a time and check which one is causing the issue and ignore it. 
There're two known ways that seems to help to fix some of this issues:
1. Use aura version 1.06.17. It seems to be the most stable one in combination with the SDK (if you're already on a newer one uninstall and clean all files/registry [ccleaner or something like that] before installing the old version.
2. If devices aren't detected or aura is just not working correctly shutdown the pc, completely cut power (disable psu or something like that) hit the power button a few times to clear power left in capacitors, wait for a few seconds and turn it back on - most of the time it works again after that.

Well I'm afraid that's all I can do (even if it's a lot of text without a real solution). 
If you have any specific problems/questions with some devices it might also be worth to join us at the [RGB.NET discord channel](https://discord.gg/9kytURv), maybe someone already solved that problem.
